ios/browser/api/bookmarks/brave_bookmarks_api.mm

```diff
@@ -622,24 +626,52 @@ - (void)removeAll {
   bookmark_model_->RemoveAllUserBookmarks();
 }
 
-- (NSArray<IOSBookmarkNode*>*)searchWithQuery:(NSString*)query
-                                     maxCount:(NSUInteger)maxCount {
-  DCHECK_CURRENTLY_ON(web::WebThread::UI);
-  DCHECK(bookmark_model_->loaded());
-  bookmarks::QueryFields queryFields;
-  queryFields.word_phrase_query.reset(
-      new std::u16string(base::SysNSStringToUTF16(query)));
-  std::vector<const bookmarks::BookmarkNode*> results;
-  GetBookmarksMatchingProperties(bookmark_model_, queryFields, maxCount,
-                                 &results);
-
-  NSMutableArray<IOSBookmarkNode*>* nodes = [[NSMutableArray alloc] init];
-  for (const bookmarks::BookmarkNode* bookmark : results) {
-    IOSBookmarkNode* node =
-        [[IOSBookmarkNode alloc] initWithNode:bookmark model:bookmark_model_];
-    [nodes addObject:node];
-  }
-  return nodes;
+- (void)searchWithQuery:(NSString*)query
+               maxCount:(NSUInteger)maxCount
+             completion:(void(^)(NSArray<IOSBookmarkNode*>*))completion {
+  auto search_with_query = [](BraveBookmarksAPI* weak_bookmarks_api,
```

Comment:
similar to above comment by @kylehickinson - was this accidentally checked in (ex: is this a change for sync bookmarks)?